db-service/test/cqn4sql/path-in-from.test.js

```diff
@@ -43,4 +43,37 @@ describe('infix filter on entities', () => {
     let query = cqn4sql(CQL`SELECT from bookshop.Books[author.ID = 22] {Books.ID}`, model)
     expect(query).to.deep.equal(CQL`SELECT from bookshop.Books as Books {Books.ID} WHERE Books.author_ID = 22`)
   })
+  it('list in where', () => {
+    const select = SELECT.from({ ref: ['bookshop.Books'] }).where([
+      { list: [{ ref: ['ID'] }] },
+      'in',
+      SELECT.from({ ref: ['bookshop.Books'] }).where([{ ref: ['ID'] }, '=', { val: 5 }]),
+    ])
+    let query = cqn4sql(select, model)
+    expect(query).to.not.be.undefined
+  })
+
+  // TO FIX
+  it('list in where in ref', () => {
+    const select = SELECT.from({
+      ref: [
+        {
+          id: 'bookshop.Books',
+          where: [
+            { list: [{ ref: ['ID'] }] },
+            'in',
+            SELECT.from({ ref: ['bookshop.Books'] }).where([{ ref: ['ID'] }, '=', { val: 5 }]),
+          ],
+        },
+      ],
+    })
+    let response
+    try {
+      response = cqn4sql(select, model)
```

Comment:
Why would we need to support that in general?
From my point of view, the `from` clause is more of a URL representation that does not allow for subselects.

Reply:
It is required to delete the compositions that exist on the table, but this is not possible when doing it directly on the view. As they might be removed and therefor having them in the path expression would fail in `cqn4sql`. 

So the idea is to map the root entries that have to be deleted from the table by using the sub select. While still using the path expression logic of `cqn4sql` to make the follow up compositions deletes.